modules/platforms/dotnet/Apache.Ignite.Core/Ignition.cs

```diff
@@ -367,6 +367,27 @@ private static void LogDiagnosticMessages(IgniteConfiguration cfg, ILogger log)
                              "To enable alternate stack check on .NET Core 3+ and .NET 5+, " +
                              "set {0} environment variable to 1.", EnvEnableAlternateStackCheck);
                 }
+
+                LogBinaryConfigurationDiagnostics(cfg.BinaryConfiguration, log);
+            }
+        }
+
+        /// <summary>
+        /// Performs binary configuration checks and logs diagnostic messages.
+        /// </summary>
+        /// <param name="cfg">Configuration.</param>
+        /// <param name="log">Log.</param>
+        internal static void LogBinaryConfigurationDiagnostics(BinaryConfiguration cfg, ILogger log)
+        {
+            if (log == null)
+                return;
+
+            if (cfg == null || !cfg.UnwrapNullablePrimitiveTypes)
+            {
+                log.Warn(
+                    $"{nameof(BinaryConfiguration)}.{nameof(BinaryConfiguration.UnwrapNullablePrimitiveTypes)} " +
+                    "is not enabled. It is recommended to enable this setting, " +
+                    "unless old behavior is required for compatibility reasons.");
```

Comment:
I'm not sure guys from CS would approve on this. Better check with @DirectXceriD just to be sure.

Reply:
Discussed in Slack, no objections from @slukyano and @DirectXceriD  .